### PR TITLE
bugfix/15921-fillcolor-hcm

### LINKED ISSUE #15921: Area Chart not displaying correct colors in high contrast mode with theme set
#### Expected behaviour
When setting the colors field in the theme object & those colors should be reflected during high contrast mode. 

i.e. the example here 
https://www.highcharts.com/docs/chart-design-and-style/themes 

The expected result of setting the highContrast theme under accessibility as that theme would be the correct colors showing up (as they would when I set those colors in chart.options.colors) 

#### Actual behaviour
Setting the colors object does not update the colors. Instead all areas are white. I do have the ability to set them to be a singular other color by setting "color" in plotOptions, but is there a separate recommended approach for setting colors in high contrast mode? I'd like to set distinct colors to maintain the same user experience. The issue here [#15040](https://github.com/highcharts/highcharts/issues/15040 ) looks related, but this is more specific to the functionality of the highContrastTheme option. 

#### Live demo with steps to reproduce
1. On windows set high contrast mode 
2. Reload chart
3. Colors not correctly set. 
https://jsfiddle.net/otc4huem/

The fiddle is just a modified version of the example fiddle provided for a stacked area chart
#### Product version
HighCharts 9.1.2

#### Affected browser(s)
Edge latest version

### PR DESCRIPTION
**Pull request by goransle**
Fixed #15921 by applying `fillColor` to series from `highContrastTheme.colors` if set. 

Also introduced `highContrastMode` option to allow disabling/forcing of high contrast mode.

WIP Fiddles:
* HCM disabled https://jsfiddle.net/goransle/vk6dLtqm/
* HCM explicitly enabled https://jsfiddle.net/goransle/7ckbvurx/

TODO:
- [x] Internal review
- [x] Add sample
- [x] Add integration test

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206562371899212

### CODE REVIEW

**Comment by KacperMadej** [comment]
- [ ] initial rendering of any Stock demo is wrongly styled and gets correct after a redraw - throws an error in the console
- [ ] color axis also throws an error (e.g. [basic map](http://utils.highcharts.local/samples/#view/maps/demo/basic-map))
- [ ] some paths with `fill:'none'` get one (`'window'`) - e.g. [flight-routes](http://utils.highcharts.local/samples/#view/maps/demo/flight-routes)
image

**Comment by goransle** [comment]
> * [ ]  some paths with `fill:'none'` get one (`'window'`) - e.g. [flight-routes](http://utils.highcharts.local/samples/#view/maps/demo/flight-routes)

Something interesting going on with `series.setOptions` here. Adding 

```js
        mappoint: {
            fillColor: 'none'
        },
        mapline: {
            fillColor: 'none'
        },
        map: {
            fillColor: 'none'
        }
```

to `HighContrastTheme.ts` produces this:
image

**Comment by goransle** [comment]
⬆️ added as issue #20716

**Review by marvin19 — CHANGES_REQUESTED** [review]
Looks good to me! I think (if possible) to remove pattern fills and find colors with enough contrast for the demos. You can use white as the color in the top triangle e.g

File: ts/Accessibility/Accessibility.ts [hunk 1]
```diff
@@ -254,9 +254,13 @@ class Accessibility {
         this.keyboardNavigation.update(kbdNavOrder);
 
         // Handle high contrast mode
+        // Should only be applied once, and not if explicitly disabled
         if (
-            !chart.highContrastModeActive && // Only do this once
-            whcm.isHighContrastModeActive()
+            !chart.highContrastModeActive &&
+            !a11yOptions.highContrastMode === false && (
```
Comment (oysteinmoseng) [inline ts/Accessibility/Accessibility.ts:260]: `a11yOptions.highContrastMode !== false`? (save 1 byte 🥳)

Comment (goransle) [reply ts/Accessibility/Accessibility.ts]: 🥳 
image

**Review by oysteinmoseng — APPROVED** [review]
🥳

File: ts/Accessibility/Accessibility.ts (hunk 1 above)
Comment (goransle) [reply ts/Accessibility/Accessibility.ts]: wait a minute...

**Review by marvin19 — APPROVED** [review]

File: ts/Accessibility/Accessibility.ts (hunk 1 above)
Comment (oysteinmoseng) [reply ts/Accessibility/Accessibility.ts]: Gzip works in mysterious ways..

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**